packages/context/src/binding-key.ts

```diff
@@ -3,7 +3,7 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-export type BindingAddress<T> = string | BindingKey<T>;
+export type BindingAddress<T = unknown> = string | BindingKey<T>;
```

Comment:
-1

It was my intention to not provide any default value for `T` generic parameter, to force all users of `BindingAddress` to explicitly provide the type.

What is your reasoning for making this change?

Reply:
Please see the difference between `BindingAddress<T=unknown>` vs. `BindingAddress<T=any>`. The 2nd case is bad as it skips the type checking but the 1st one is good as it gives us the flexibility to not provide the type if it does NOT matter any way.

Reply:
I am aware of the difference between `BindingAddress<T=unknown>` and `BindingAddress<T=any>`.

I am concerned that by allowing BindingAddress with no template argument, we make it too easy for developers to be lazy and exclude the type argument even in places where it should be used. For example, inside templated functions/classes that work with any `T` type, it's all too easy to use `BindingAddress` instead of more correct `BindingAddress<T>`.

Reply:
With `T` default to `unknown`, I don't think it will sacrifice type checking. See explanation at https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html.

Reply:
I think you are not getting my point. Let's not worry about this until a real world usage proves whether I was correct in my concerns.